source/slang/slang-ir-lower-existential.cpp

```diff
@@ -105,7 +105,16 @@ namespace Slang
             auto builder = &builderStorage;
             builder->setInsertBefore(extractInst);
 
-            auto element = extractTupleElement(builder, extractInst->getOperand(0), elementId);
+            IRInst* element = nullptr;
+            if (extractInst->getOperand(0)->getDataType()->findDecoration<IRComInterfaceDecoration>())
+            {
+                // If this is an COM interface, the elements (witness table/rtti) are just the interface value itself.
+                element = extractInst->getOperand(0);
```

Comment:
This *maybe* warrants a `TODO:` comment about how eventually we might define that the witness table is found by treating the object as a pointer to the witness table (so that the Slang "witness table" is identical to the COM/C++ "virtual function table").